test_data/pil/expand_fixed.pil

```diff
@@ -0,0 +1,172 @@
+            };
+
+namespace std::array;
+    /// This is a built-in function taking an array argument and returning
+    /// the length of the array.
+    /// This symbol is not an empty array, the actual semantics are overridden.
+    let<T> len: T[] -> int = [];
+
+    /// Evaluates to the array [f(0), f(1), ..., f(length - 1)].
+    let<T> new: int, (int -> T) -> T[] = |length, f| std::utils::fold(length, f, [], |acc, e| (acc + [e]));
+
+    /// Returns a new array equal to arr except that the element at index i is x.
+    let<T> set_element: T[], int, T -> T[] = |arr, i, x| {
+        let _ = std::check::assert(i < len(arr), || "Index out of bounds");
+        map_enumerated(arr, |j, y| if i == j { x } else { y })
+    };
+
+    /// Returns a new array of length l containing the elements of arr starting at index start.
+    let<T> sub_array: T[], int, int -> T[] = |arr, start, l| std::array::new(l, |i| arr[start + i]);
+
+    /// Evaluates to the array [f(arr[0]), f(arr[1]), ..., f(arr[len(arr) - 1])].
+    let<T1, T2> map: T1[], (T1 -> T2) -> T2[] = |arr, f| new(len(arr), |i| f(arr[i]));
+
+    /// Evaluates to the array [f(0, arr[0]), f(1, arr[1]), ..., f(len(arr) - 1, arr[len(arr) - 1])].
+    let<T1, T2> map_enumerated: T1[], (int, T1 -> T2) -> T2[] = |arr, f| new(len(arr), |i| f(i, arr[i]));
+
+    /// Computes folder(...folder(folder(initial, arr[0]), arr[1]) ..., arr[len(arr) - 1])
+    let<T1, T2> fold: T1[], T2, (T2, T1 -> T2) -> T2 = |arr, initial, folder| std::utils::fold(len(arr), |i| arr[i], initial, folder);
+
+    /// Returns the sum of the array elements.
+    // TODO: Should make use of the Default or Zero trait instead of FromLiteral (then we can also
+    // use this function to flatten an array of arrays.
+    let<T: Add + FromLiteral> sum: T[] -> T = |arr| fold(arr, 0, |a, b| a + b);
+
+    /// Returns the product of the array elements.
+    let<T: Mul + FromLiteral> product: T[] -> T = |arr| fold(arr, 1, |a, b| a * b);
+
+    /// Zips two arrays
+    /// TODO: Assert that lengths are equal when expressions are supported.
+    let<T1, T2, T3> zip: T1[], T2[], (T1, T2 -> T3) -> T3[] = |array1, array2, fn| new(len(array1), |i| fn(array1[i], array2[i]));
+
+namespace Main(8);
+
+    // A ArrayTerm in an `ArrayExpression`
+    enum ArrayTerm {
+        Repeat(int[]),
+        Once(int[])
+    }
+
+    // returns the total size of the repeated array in this `ArrayExpression`
+    let solve: ArrayTerm[] -> int = |terms| {
+        let (_, total_size_of_non_repeated) = std::array::fold(terms, (std::prelude::false, 0), |(found_repeated, len), ArrayTerm| {
+            match ArrayTerm {
+                // we can have `[]*` as many times as we want
+                ArrayTerm::Repeat([]) => (found_repeated, len),
+                ArrayTerm::Repeat(a) => {
+                    if found_repeated {
+                        std::check::panic("unsolvable because two terms are repeated");
+                        (found_repeated, 42)
```

Comment:
I think this should work because panic returns `!`
```suggestion
                        std::check::panic("unsolvable because two terms are repeated")
```